src/internal/packager/images/pull.go

```diff
@@ -448,40 +486,13 @@ func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Im
 		// If crane is unable to pull the image, try to load it from the local docker daemon.
 		message.Debugf("crane unable to pull image %s: %s", src, err)
 		spinner.Updatef("Falling back to docker for %s. This may take some time.", src)
-
-		// Parse the image reference to get the image name.
-		reference, err := name.ParseReference(src)
-		if err != nil {
-			return nil, false, fmt.Errorf("failed to parse image reference %s: %w", src, err)
-		}
-
-		// Attempt to connect to the local docker daemon.
-		ctx := context.TODO()
-		cli, err := client.NewClientWithOpts(client.FromEnv)
-		if err != nil {
-			return nil, false, fmt.Errorf("docker not available: %w", err)
-		}
-		cli.NegotiateAPIVersion(ctx)
-
-		// Inspect the image to get the size.
-		rawImg, _, err := cli.ImageInspectWithRaw(ctx, src)
-		if err != nil {
-			return nil, false, fmt.Errorf("failed to inspect image %s via docker: %w", src, err)
-		}
-
-		// Warn the user if the image is large.
-		if rawImg.Size > 750*1000*1000 {
-			warn := pterm.DefaultParagraph.WithMaxWidth(message.TermWidth).Sprintf("%s is %s and may take a very long time to load via docker. "+
-				"See https://docs.zarf.dev/docs/faq for suggestions on how to improve large local image loading operations.",
-				src, utils.ByteFormat(float64(rawImg.Size), 2))
-			spinner.Warnf(warn)
+		var localErr error
+		img, localErr = i.pullLocalImage(src, spinner)
+		if localErr != nil {
+			// if local failed, return the original remote error
+			return nil, false, err
```

Comment:
This error could use massaging now that we are showing it to all users if we can - at least to also say that it could not be found by docker since we actually don't know which error the user was expecting to see (it's most likely it would be from a remote but the error may confuse them if it were the opposite)